compiler/z/codegen/UnaryEvaluator.cpp

```diff
@@ -310,21 +310,46 @@ OMR::Z::TreeEvaluator::dsqrtEvaluator(TR::Node * node, TR::CodeGenerator * cg)
    {
    TR::Node * firstChild = node->getFirstChild();
    TR::Register * targetRegister = NULL;
-   TR::Register * opRegister = cg->evaluate(firstChild);
 
-   if (cg->canClobberNodesRegister(firstChild))
+
+   // Calculate it for ourselves
+   if (firstChild->getOpCode().isLoadConst())
       {
-      targetRegister = opRegister;
+      union { double valD; int64_t valI; } result;
+      targetRegister = cg->allocateRegister(TR_FPR);
+      result.valD = sqrt(firstChild->getDouble());
+      TR::S390ConstantDataSnippet * cds = cg->findOrCreate8ByteConstant(node, result.valI);
+      generateRXInstruction(cg, TR::InstOpCode::LD, node, targetRegister, generateS390MemoryReference(cds, cg, 0, node));
       }
    else
       {
-      targetRegister = cg->allocateRegister(TR_FPR);
+      TR::Register * opRegister = NULL;
```

Comment:
`opRegister` seems to be used only within the `else` block below. Can we move the declaration there to limit the scope?